### PR TITLE
VizTooltips: Tolerate missing annotation text

### PR DESCRIPTION
**Pull request by darrenjaneczek**
**What is this feature?**

Prevents a `TypeError` by allowing the resulting `text` to be undefined.
Later on in the code, `text` is only used if it is a string type.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by leeoniya** [comment]
does this need the backport?

this is gated behind `newVizTooltips`, which we don't have enabled by default and are just starting to roll out in `ops`: https://github.com/grafana/deployment_tools/pull/124451

File: public/app/plugins/panel/timeseries/plugins/annotations2/AnnotationTooltip2.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const AnnotationTooltip2 = ({ annoVals, annoIdx, timeZone, onEdit }: Prop
     });
 
   let time = timeFormatter(annoVals.time[annoIdx]);
-  let text = annoVals.text[annoIdx];
+  let text = annoVals.text?.[annoIdx];
```
[the remark this answers is not in the document]
Comment (leeoniya) [reply public/app/plugins/panel/timeseries/plugins/annotations2/AnnotationTooltip2.tsx:35]: i guess we can also omit the `<br />` below when `=== ''`

**Comment by baldm0mma** [comment]
@leeoniya @darrenjaneczek Grafana's current back port policy only supports back porting if the change is considered either a "critical bug fix" or "critical security patch." This doesn't fall into either category. Nevertheless, this is a thoughtful contribution, so should still be considered for G11. I'll update the labels accordingly.

**Comment by darrenjaneczek** [comment]
Thanks.

Agreed, no backport needed.

Applied recommendations.

**Review by leeoniya — APPROVED** [review]
LGTM, thanks!

**Review by baldm0mma — APPROVED** [review]
LGTM.

**Pull request merged**